### PR TITLE
don't record error when request a non-exist user

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: routers/repo/http.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func httpBase(ctx *context.Context) (h *serviceHandler) {
 			if redirectUserID, err := models.LookupUserRedirect(username); err == nil {
 				context.RedirectToUser(ctx, username, redirectUserID)
 			} else {
-				ctx.NotFound("GetUserByName", err)
+				ctx.NotFound(fmt.Sprintf("User %s is not exist", username), nil)
```
Comment (zeripath) [inline routers/repo/http.go]:
```suggestion
				ctx.NotFound(fmt.Sprintf("User %s does not exist", username), nil)
```

**Comment by zeripath** [comment]
If there a particular common user that is being reported because e.g. serviceworker.js then applying this will mean that we're hiding an actual performance problem. For example - we shouldn't be trying to get serviceworker.js from the db in the first place as it's an invalid username.

**Comment by lunny** [comment]
> If there a particular common user that is being reported because e.g. serviceworker.js then applying this will mean that we're hiding an actual performance problem. For example - we shouldn't be trying to get serviceworker.js from the db in the first place as it's an invalid username.

I think some router tests should resolve the problem. This PR is to prevent to log code stack when a non-exist user.

**Review by zeripath — APPROVED** [review]

**Review by KN4CK3R — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Pull request merged**